### PR TITLE
test: port library tests and add type safety

### PR DESCRIPTION
**Pull request by adamegyed**
## Motivation

We ported the libraries from the RI, but not the corresponding tests. As we may make changes to the libraries during the optimization process, we should also port the tests.

Additionally, to make sure we're not doing anything unexpected with the value returned as a "validation flags" `uint8` in ValidationConfigLib, we can wrap it in a user-defined value type. This is not used the the account itself, so it does not change performance characteristics.

## Solution

Port the tests and add a user-defined value type for validation flags.

### CODE REVIEW

File: src/libraries/ValidationConfigLib.sol [hunk 1]
```diff
@@ -3,6 +3,8 @@ pragma solidity ^0.8.26;
 
 import {ModuleEntity, ValidationConfig} from "@erc6900/reference-implementation/interfaces/IModularAccount.sol";
 
+type ValidationFlags is uint8;
+
```
Comment (fangting-alchemy) [inline src/libraries/ValidationConfigLib.sol:12]:
```suggestion
// Validation flags layout:
// 0b00000___ // unused
// 0b_____A__ // isGlobal
// 0b______B_ // isSignatureValidation
// 0b_______C // isUserOpValidation
type ValidationFlags is uint8;

```

Comment (fangting-alchemy) [reply src/libraries/ValidationConfigLib.sol:12]: it will cost 3 gas (I know it is not much) to wrap and unwrap. Is it worth it to define a type?

Comment (adamegyed) [reply src/libraries/ValidationConfigLib.sol:12]: In theory, user-defined value types are free, from a gas perspective, because they are "zero cost abstractions" - aka type checks that only happen during compile time, but don't change anything at execution. https://docs.soliditylang.org/en/latest/types.html#user-defined-value-types

However, like I mentioned in Slack #eng-solidity, I think I found an edge case where it affects the optimizer. But, this PR doesn't use this type in the source files anywhere, just in tests, so it's more of a footgun prevention mechanism than a functionality change.

Comment (fangting-alchemy) [reply src/libraries/ValidationConfigLib.sol:12]: TIL on UDVT.

Does this change affect gas benchmarks? Why only 4 actions run on this PR?

Comment (adamegyed) [reply src/libraries/ValidationConfigLib.sol:12]: > Does this change affect gas benchmarks?

No, they're the same

> Why only 4 actions run on this PR?

The other actions only happen when the PR is against `develop`, this is still a stacked branch (for now)

**Review by fangting-alchemy — APPROVED** [review]
